tests/unit/components/AddObsModal/AddObsButton.test.js

```diff
@@ -0,0 +1,69 @@
+import { screen } from "@testing-library/react-native";
+import AddObsButton from "components/AddObsModal/AddObsButton";
+import React from "react";
+import * as useCurrentUser from "sharedHooks/useCurrentUser.ts";
+import { zustandStorage } from "stores/useStore";
+import factory from "tests/factory";
+import { renderComponent } from "tests/helpers/render";
+
+// Mock getCurrentRoute to return ObsList
+jest.mock( "navigation/navigationUtils", () => ( {
+  getCurrentRoute: () => ( {
+    name: "ObsList"
+  } )
+} ) );
+
+const mockUser = factory( "LocalUser" );
+
+jest.mock( "sharedHooks/useCurrentUser", () => ( {
+  __esModule: true,
+  default: () => undefined
+} ) );
+
+describe( "AddObsButton", () => {
+  it( "should not have accessibility errors", () => {
+    const addObsButton = <AddObsButton />;
+
+    expect( addObsButton ).toBeAccessible();
+  } );
+
+  it( "renders correctly", () => {
+    renderComponent( <AddObsButton /> );
+    // Snapshot test
+    expect( screen ).toMatchSnapshot();
+  } );
+
+  it( "does not render tooltip in default state", () => {
+    renderComponent( <AddObsButton /> );
+
+    const tooltipText = screen.queryByText(
+      "Press and hold to view more options"
+    );
+    expect( tooltipText ).toBeFalsy();
+  } );
+} );
+
+describe( "shows tooltip", () => {
```

Comment:
It looks like you're skipping a test for the `justFinishedSignup` state here. Any reason for that?